polkadot/runtime/rococo/src/lib.rs

```diff
@@ -2052,7 +2054,9 @@ sp_api::impl_runtime_apis! {
 			list_benchmarks!(list, extra);
 
 			let storage_info = AllPalletsWithSystem::storage_info();
-			return (list, storage_info)
+			let max_extrinsic_weight = BlockWeights::get().per_class.get(DispatchClass::Normal).max_extrinsic.unwrap();
+			let db_weight: frame_support::weights::RuntimeDbWeight = <Self as frame_system::Config>::DbWeight::get();
+			(list, storage_info, max_extrinsic_weight, db_weight)
```

Comment:
Creating a single struct for `Weight` and `RuntimeDbWeight` is either way a better solution. Would you prefer to have it included in `BenchmarkList` / `StorageInfo` and make it backwards compatible or add an additional parameter which results in a breaking change?

Reply:
I think it would need one breaking change, yes. But from then on we can avoid further breaking changes by using the following pattern:  
```rust
struct AllBenchmarkData {
  list: Vec<frame_benchmarking::BenchmarkList>,
  info: Vec<frame_support::traits::StorageInfo>,
  max_ext_weight: Option<..>,
  db_weights: Option<..>
}
```
and having `Default` on that thing. The runtime devs can use `..Default::default()` to stay forward compatible and we can keep adding `Option`s without directly breaking their code.